### PR TITLE
feat: migrate Segment utils [split of 8387 - 2]

### PR DESCRIPTION
**Pull request by NicolasMassart**
## **Description**

>[!NOTE]
> #8387 is too big for review, splitting it into contextual/domain scoped smaller PRs.
> you can refer to #8387 for global view

> [!WARNING]
> This PR depends on PR slice 1 #8573, currently targeting `feat/1461_migrate_segment_1_core`  branch to allow build. Will require change once branch merged in `feat/batch_1129_segment`

### This is the part of #8387 related to the utils code of the Segment work.

update metrics utils
- rename trackAfterInteractions to trackOnboarding to match purpose
- update for synchronous getInstance
- fix types
- update exports
- move and update trackErrorAsAnalytics outside of Analyticsv2

## **Related issues**

Related to MetaMask/mobile-planning/issues/1461

## **Manual testing steps**

NA

### **Before**

NA

### **After**

NA

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by Cal-L — CHANGES_REQUESTED** [review]
Nice - love how this is split out. For folder structure and naming, I think there's a more scalable way that aligns more with how our current files are setup. The design system components set it up in this manner as well. The final structure + naming would look like so:

image


LMK what you think.

File: app/util/metrics/index.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import DeviceAnalyticsMetaData from './DeviceAnalyticsMetaData/generateDeviceAnalyticsMetaData';
 import UserSettingsAnalyticsMetaData from './UserSettingsAnalyticsMetaData/generateUserProfileAnalyticsMetaData';
-import TrackAfterInteractions from './TrackAfterInteraction/trackAfterInteractions';
-
+import TrackError from './TrackError/trackErrorAsAnalytics';
 export default DeviceAnalyticsMetaData;
```
Comment (Cal-L) [inline app/util/metrics/index.ts:4]: Is there a reason why we aren't reexporting TrackOnboarding as well? Could we also consolidate the exports?

**Review by Cal-L — COMMENTED** [review]
Left some comments

File: app/util/metrics/index.ts (hunk 1 above)
Comment (NicolasMassart) [reply app/util/metrics/index.ts:4]: Will update these imports in another PR, trying to reduce conflicts and merge here

File: app/util/metrics/DeviceAnalyticsMetaData/DeviceAnalyticsMetaData.types.ts
Comment (Cal-L) [inline app/util/metrics/DeviceAnalyticsMetaData/DeviceAnalyticsMetaData.types.ts:1]: Let's follow the folder structure similar to how Design system team has theirs set up. Your `useMetrics` hook is set up this way as well. Here's an example of how `metrics` directory might look like after the improvement.:

image

**Review by Cal-L — COMMENTED** [review]
Left a comment

**Comment by NicolasMassart** [comment]
Closed as this was reworked and split differently in smaller PRs for the migration

**Pull request closed without merging**